src/main/java/uk/gov/di/ipv/cri/common/library/service/AuditEventFactory.java

```diff
@@ -38,20 +41,48 @@ <T> AuditEvent<T> create(String eventType, AuditEventContext auditEventContext,
                         eventPrefix + "_" + eventType,
                         issuer);
         if (Objects.nonNull(auditEventContext)) {
-            if (Objects.nonNull(auditEventContext.getPersonIdentity())) {
-                auditEvent.setRestricted(auditEventContext.getPersonIdentity());
-            }
+            Map<String, String> requestHeaders = auditEventContext.getRequestHeaders();
+
+            auditEvent.setRestricted(
+                    createAuditEventResricted(
+                            requestHeaders, auditEventContext.getPersonIdentity()));
+
             auditEvent.setUser(
-                    createAuditEventUser(
-                            auditEventContext.getRequestHeaders(),
-                            auditEventContext.getSessionItem()));
+                    createAuditEventUser(requestHeaders, auditEventContext.getSessionItem()));
         }
         if (Objects.nonNull(extensions)) {
             auditEvent.setExtensions(extensions);
         }
         return auditEvent;
     }
 
+    @SuppressWarnings("deprecation")
+    private PersonIdentityDetailed createAuditEventResricted(
+            Map<String, String> requestHeaders, PersonIdentityDetailed personIdentityDetails) {
+        PersonIdentityDetailed restricted = null;
+
+        if (Objects.nonNull(personIdentityDetails)) {
+            restricted =
+                    new PersonIdentityDetailed(
+                            personIdentityDetails.getNames(),
+                            personIdentityDetails.getBirthDates(),
+                            personIdentityDetails.getAddresses(),
+                            personIdentityDetails.getDrivingPermits(),
+                            personIdentityDetails.getPassports());
+        }
+
+        if (requestHeaders.containsKey(TXMA_AUDIT_ENCODED)) {
+            if (Objects.isNull(restricted)) {
+                restricted = new PersonIdentityDetailed(null, null, null, null, null);
+            }
+            DeviceInformation deviceInformation = new DeviceInformation();
+            deviceInformation.setEncoded(requestHeaders.get(TXMA_AUDIT_ENCODED));
+            restricted.setDeviceInformation(deviceInformation);
+        }
+
+        return restricted;
+    }
+
```

Comment:
Maybe 
```
 private PersonIdentityDetailed createAuditEventResricted(
            Map<String, String> requestHeaders, PersonIdentityDetailed personIdentityDetails) {
        PersonIdentityDetailed restricted = null;

        if (Objects.nonNull(personIdentityDetails)) {
            restricted = personIdentityDetails;
        }

        if (requestHeaders.containsKey(TXMA_AUDIT_ENCODED)) {
            if (Objects.isNull(restricted)) {
                restricted = PersonIdentityDetailedBuilder.builder().build();
            }
            DeviceInformation deviceInformation = new DeviceInformation();
            deviceInformation.setEncoded(requestHeaders.get(TXMA_AUDIT_ENCODED));
            restricted.setDeviceInformation(deviceInformation);
        }

        return restricted;
    }
```

Reply:
I believe this can be further simplified, by extending the with a `withDeviceInformation` method

Reply:
Could you please elaborate on that? What are you thinking the `withDeviceInformation` would do? Does this follow a specific design pattern?

Reply:
Made the restricted = personIdentityDetails; change!

Reply:
Actually `restricted = new PersonIdentityDetailed(null, null, null, null, null);` is fine  since it use is within the library Only consumers of the library require the Builder or the Factory

Do we really need the following? We would end up returning `restricted = null` if it isn't `TXMA_AUDIT_ENCODED` any way

```
       PersonIdentityDetailed restricted = null;

        if (Objects.nonNull(personIdentityDetails)) {
            restricted = personIdentityDetails;
        }
```

Maybe 

```
     PersonIdentityDetailed restricted = personIdentityDetails;
```

or maybe get rid of the restricted variable and reuse 

```
    personIdentityDetails = new PersonIdentityDetailed(null, null, null, null, null);
```

and then `return personIdentityDetails` at the end

Test coverage is needed for instance passing in a null `personIdentityDetails` when `TXMA_AUDIT_ENCODED`